proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -394,14 +390,23 @@ message Checksum {
     bytes value = 3;
 }
 
-message CallbackInfo {
-    // The namespace failover version at the time this callback info was updated.
-    // Used for conflict resolution when syncing state across clusters.
+// Keeps track of the ranges of transition counts per namespace failover version.
+// Each task generated by the HSM framework is imprinted with the current transaction’s `NamespaceFailoverVersion` and
+// `MaxTransitionCount` at the end of the transaction.
+// When a task is being processed, the `StateTransitionHistory` is compared with the imprinted task information to
+// verify that a task is not referencing a stale state or that the task itself is not stale.
+// For example, if the state has a history of `[{v: 1, t: 3}, {v: 2, t: 5}]`, task A `{v: 2, t: 4}` **is not**
+// referencing stale state because for version `2` transitions `4-5` are valid, while task B `{v: 2, t: 6}` **is**
+// referencing stale state because the transition count is out of range for version `2`.
+// Furthermore, task C `{v: 1, t: 4}` itself is stale because it is referencing an impossible state, likely due to post
+// split-brain reconciliation.
+message VersionedTransition {
+    // The namespace failover version at transition time.
     int64 namespace_failover_version = 1;
-    // Number of transitions on this callback object. Used to verify that a task
-    // is not referencing a stale callback object or that the task itself is not stale.
-    int32 transition_count = 2;
-    // ID of this callback, used as a key in the mutable state map.
-    string id = 3;
-    temporal.api.workflow.v1.CallbackInfo public_info = 4;
+    // Maximum state transitions count for the given version.
+    int64 max_transition_count = 2;
```

Comment:
Is this the _maximum_ state transition perceived during the specified namespace_failover_version? The terminology here is confusing

Reply:
Yes. I'll try to clarify the comment.

Reply:
Done.